src/qbittorrent.exe.manifest

```diff
@@ -1,45 +1,39 @@
-                        processorArchitecture="*"
-                        publicKeyToken="6595b64144ccf1df"
-                        language="*" />
-    </dependentAssembly>
-  </dependency>
+    <!-- Declare Windows version support (currently: 7, 8, 8.1, and 10) -->
+    <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+        <application>
+            <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
+            <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
+            <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
+            <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" />
+        </application>
+    </compatibility>
 
-  <!-- Indicate UAC compliance, with no need for elevated privileges (Win Vista and later) -->
-  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
-    <security>
-      <requestedPrivileges>
-        <requestedExecutionLevel level="asInvoker" uiAccess="false" />
-      </requestedPrivileges>
-    </security>
-  </trustInfo>
+    <!-- Enable use of version 6 of the common controls -->
+    <dependency>
+        <dependentAssembly>
+            <assemblyIdentity type="win32" name="Microsoft.Windows.Common-Controls" version="6.0.0.0" processorArchitecture="*" publicKeyToken="6595b64144ccf1df" language="*" />
+        </dependentAssembly>
+    </dependency>
 
-  <!-- Declare support for various versions of Windows -->
-  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
-      <application>
-          <!-- Windows Vista -->
-          <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}" />
-          <!-- Windows 7 -->
-          <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
-          <!-- Windows 8 -->
-          <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
-          <!-- Windows 8.1 -->
-          <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
-          <!-- Windows 10 -->
-          <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" />
-      </application>
-  </compatibility>
-  <!-- Enable long paths that exceed MAX_PATH in length -->
-  <asmv3:application>
-    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">
-      <longPathAware>true</longPathAware>
-    </asmv3:windowsSettings>
-  </asmv3:application>
+    <!-- Indicate UAC compliance, with no need for elevated privileges -->
+    <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+        <security>
+            <requestedPrivileges>
+                <requestedExecutionLevel level="asInvoker" uiAccess="false" />
+            </requestedPrivileges>
+        </security>
+    </trustInfo>
+
+    <!-- Set important attributes for better application behavior on modern systems -->
+    <asmv3:application>
+        <asmv3:windowsSettings>
+            <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
```

Comment:
Seems it still doesn't affect qBittorrent.
Our own code uses only Unicode variant of Win32 API.
I dumped official Qt binaries, and it confirmed my belief that it doesn't use ANSI Win32 API either. In addition, it is really unprofitable for Qt to use ANSI Win32 API (even with UTF8 code page), since it itself uses UTF16 to encode strings, so it would require frequent transcoding between UTF8 (or any other 1 byte encoding) and UTF16.

Reply:
@glassez still, since there is no downside, might as well leave it like this since it is the recommended practice, no?

Reply:
>since it is the recommended practice, no?

I think not, IMO the real recommended practice would be MS set it as the default, not telling/urging users to opt-in.

qbt has always worked fine without it and adding this directive which we have no need and no effect sounds like redundant (or technical debt) for me.
Same viewpoint on other directives you added, you should propose them **only if** they have **real** impacts/effects, not because they are new or shiny/trendy or do some *suspected* fix.
At least separate them to its own commits and add appropriate comments/rational for them.

Reply:
@Chocobo1

OK, I will separate each into separate commits.

>I think not, IMO the real recommended practice would be MS set it as the default, not telling/suggesting users to opt-in.

This logic is flawed. Not sure what the defaults for manifest generation in VS actually are, but just because some "old" value is the default does not mean it is the recommended practice, or that the "new" values are reserved for "bleeding edge" cases, especially in ecosystems where backwards compatibility is a core design goal, such as is the case for Windows. The docs make it pretty clear what the recommended practices are. If you don't want these changes, enjoy receiving endless bug reports about broken multi-monitor support due to not setting PerMonitorV2, which is a prerequisite for fixing it on modern systems.

About each individual addition:

- `dpiAware`/`dpiAwareness`: This has been discussed at length in the past, especially with @xavier2k6 (https://github.com/qbittorrent/qBittorrent/projects/2#column-12115743). These changes are required for proper multi-monitor support.
- `activeCodePage`: This is recommended by the docs, and only "affects" Windows 10 >= 1903 anyway (https://docs.microsoft.com/en-au/windows/win32/sbscs/application-manifests#activecodepage, https://docs.microsoft.com/en-us/windows/uwp/design/globalizing/use-utf8-code-page). Not keeping up with these recommendations is what accrues technical debt, not adopting them. Adopting such changes prevents potential issues down the line, if it does not explicitly "fix" anything now.
- `heapType`: similar to the previous point. This is the new modern way (https://docs.microsoft.com/en-au/windows/win32/sbscs/application-manifests#heaptype), and only applies to Windows 10 >= 2004.

> Also "Reorganize and reformat Windows manifest" commit looks meaningless to me, I vote to drop it.

It simply formats the file with 4 space indentation, similar to the main source files, which is more consistent, and moves the `<compatibility>` stanza to the top, which is the most important one. I don't think these are bad changes. Are you sure?

Reply:
>`activeCodePage`: This is recommended by the docs

Seems "you heard bells, but you didn't know where the sound was coming from".
Where exactly in the documentation is it strictly recommended for unconditional use in all cases? It just mentions that this might be a good idea in general, in case you're dealing with UTF-8 encoded strings, and the application uses ANSI API.
But this does not apply well to qBittorrent (as, indeed, to any Qt application). Although we're dealing with UTF-8 encoded strings passed from some outer sources we still have convert them to QString (i.e. in UTF-16) in the vast majority of cases. So why we need to convert them back to UTF-8 when calling Win32 API instead of using its Unicode (UTF-16) variant? Especially considering the fact that Windows operates with UTF-16 natively.
If `activeCodePage` doesn't affect us, why should we clog our Manifest with it?
>Adopting such changes prevents potential issues down the line, if it does not explicitly "fix" anything now.

`activeCodePage` option can neither prevent nor fix anything by itself. It just allows you to use UTF-8 (or other code page) in your application regardless of system one (if you use ANSI Win32 API).

Reply:
>This logic is flawed. 

Sorry, but it works well until proven otherwise.

>but just because some "old" value is the default does not mean it is the recommended practice

I never said that.

>or that the "new" values are reserved for "bleeding edge" cases

Yes it is. If it is not then why don't they enable it by default and benefit everyone?

>If you don't want these changes, enjoy...

Whatever you say.... try show some real confirmations from users, you have proven nothing yet.

>I don't think these are bad changes. Are you sure?

Yes, I won't be pointing out if I don't care.

Reply:
@Chocobo1 

> Sorry, but it works well until proven otherwise.

The logic of the statement "I think not, IMO the real recommended practice would be MS set it as the default, not telling/suggesting users to opt-in." is flawed at fundamental theoretical level and goes against what is written in the docs. There is nothing to be "proven otherwise". They _do_ recommend switching, they just don't do it _by default_ because of backwards compatibility/legacy users. Similar case to `/Zc:__cplusplus`.

> I never said that.

It is the logical consequence of this, which you did say:

> I think not, IMO the real recommended practice would be MS set it as the default, not telling/suggesting users to opt-in.

---

> Yes it is. If it is not then why don't they enable it by default and benefit everyone?

Because of backwards compatibility/legacy users in the wider Windows ecosystem. Read my previous post again. We've been over this in previous instances, such as with `/Zc:__cplusplus`.

> Whatever you say.... try show some real confirmations from users that your proposed changes worked, until now you have proven nothing yet.

From reading the documentation it should be obvious to you that these changes are a net benefit, _regardless of whether you can already see a "confirmation" right now_. But, since you insist so much: https://github.com/qbittorrent/qBittorrent/issues/15019#issuecomment-851013929.

Now the burden of proof is on you to prove how any of this is a regression or breaks something.

> And don't forget to put them in the commit comment.

:+1:

Reply:
Fixed.